module/surveys/helper.js

```diff
@@ -0,0 +1,1069 @@
+
+                    status = messageConstants.apiResponses.SURVEY_ALREADY_EXISTS
+
+                } else {
+                    
+                    let survey = {}
+
+                    if (userOrganisations.createdFor) {
+                        survey["createdFor"] = userOrganisations.createdFor;
+                    }
+                    if (userOrganisations.rootOrganisations) {
+                        survey["rootOrganisations"] = userOrganisations.rootOrganisations;
+                    }
+
+                    survey["status"] = messageConstants.common.PUBLISHED;
+                    survey["deleted"] = "false";
+                    survey["solutionId"] = solution._id;
+                    survey["solutionExternalId"] = solution.externalId;
+                    survey["createdBy"] = userId;
+                    survey["startDate"] = date;
+                    survey["endDate"] = solution.endDate ? solution.endDate : date.setFullYear(date.getFullYear() + 1);
+                    survey["name"] = solution.name;
+                    survey["description"] = solution.description;
+                    
+                    if (solution.programId) {
+                       survey["programId"] = solution.programId;
+                    }
+                    if (solution.programExternalId) {
+                       survey["programExternalId"] = solution.programExternalId;
+                    }
+                         
+                    surveyDocument = await this.create(
+                        survey
+                    );
+
+                    if (surveyDocument._id) {
+                        surveyId = surveyDocument._id
+                    }
+
+                    surveyId ? status = `${surveyId._id} created` : status = `${surveyId._id} could not be created`;
+                }
+
+                return resolve({
+                    success: true,
+                    data: {
+                        status : status,
+                        _id: surveyId
+                    }
+                })
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        })
+    }
+
+
+    /**
+     * Send user notifications. 
+     * @method
+     * @name sendUserNotifications
+     * @param {Object} [surveyData = {}] - .
+     * @param {String} [userId = ""] - logged in user id.      
+     * @returns {Object} message and success status.
+    */
+
+     static sendUserNotifications(userId = "", surveyData = {}) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+
+                const kafkaMessage = await kafkaClient.pushEntityAssessorNotificationToKafka({
```

Comment:
@deepakr123 - Rename pushEntityAssessorNotificationToKafka it to pushUserMappingNotificationToKafka

Reply:
resolved